Issue #2871632 port the base settings config form to Drupal 8

m4olivei opened the pull request:
Ports the base module settings for to Drupal 8.

Also a bunch of code style updates in here, namely changing to PHP short array syntax.

Things to check:
 - [ ] All settings ported over.
 - [ ] Test the admin form, change all settings. Try inputting invalid data for the ads fields to ensure validation works.
 - [ ] Save a FB page_id and check the source on the home page to ensure the metatag renders.

deviantintegral commented on fb_instant_articles.module:
```diff
@@ -8,6 +8,38 @@
 use Drupal\Core\Routing\RouteMatchInterface;
 
 /**
+ * No ads will automatically be inserted into Instant Articles.
```
Can we put these constants in a class somewhere? Same thing with the method below.

bgbohling replied:
@deviantintegral Is there a convention to follow for this, like class name, type, etc.?

m4olivei replied:
@deviantintegral we can, but at this stage of the port there isn't a good home for it yet. Let's leave it for now and come back to it. These constants will most likely see actual use in https://www.drupal.org/node/2872807. There should be a good home in the implementation there. I'd like to wait until that evolves, in the meantime, I'll mark a @ todo.

deviantintegral commented on fb_instant_articles.module:
```diff
@@ -46,24 +116,24 @@ function fb_instant_articles_page_attachments_alter(array &$page) {
-  $appender = array(
+  $enable_transformer_logging = \Drupal::config('fb_instant_articles.base_settings')->get('enable_logging');
+  $appender = [
     'class' => $enable_transformer_logging ? '\Drupal\fb_instant_articles\DrupalLoggerAppender' : 'LoggerAppenderNull',
```
Double check, is `LogAppenderNull` supposed to be a full class name, or is it just a special keyword?

bgbohling replied:
LoggerAppenders are a FB SDK thing that provides a way to forward log messages from the SDK to elsewheres, like Drupal watchdog.  So I'm going to guess that `LoggerAppenderNull` is a keyword.

m4olivei replied:
It is a class name. It lives in the apache/log4php package. It's in the global namespace, and this value is used within the package, so I believe it's correct as is.

deviantintegral replied:
Boo, global namespace. Should it be prefixed with a \ then to make it clear, given that the other option is namespaced?

deviantintegral commented on src/Form/BaseSettingsForm.php:
```diff
@@ -0,0 +1,263 @@
+      '#type' => 'textfield',
+      '#default_value' => $config->get('page_id'),
+      '#required' => TRUE,
+      '#description' => $this->t('In order to designate the domain that will host your
```
I'm pretty sure this has to all be on one line for localize.drupal.org to be able to extract the string.

bgbohling replied:
Made it a single line.

deviantintegral commented on src/Form/BaseSettingsForm.php:
```diff
@@ -0,0 +1,263 @@
+      $form_state->setErrorByName('ads_iframe_url', $this->t('You must specify a valid source URL for your Ads when using the Source URL ad type.'));
+    }
+
+    if (filter_var($ads_iframe_url, FILTER_VALIDATE_URL) === FALSE) {
```
We should probably use `\Drupal\Component\Utility\UrlHelper::isValid` here.

bgbohling replied:
OK, using that thing, then.

deviantintegral commented on src/Form/BaseSettingsForm.php:
```diff
@@ -0,0 +1,263 @@
+    }
+
+    if (preg_match('/^[\d_]+$/', $ads_an_placement_id) !== 1) {
+      $form_state->setErrorByName('ads_an_placement_id', $this->t('You must specify a valid Placement ID when using the Audience Network ad type.'));
```
Does it help to call out what the placement ID is or where you can find it?

bgbohling replied:
It looks like this is something set up in the brand's Audience Network account for Instant Articles.  Adding the instrux from the FB developer docs to the form feedback.

bgbohling commented:
D'oh!  Forgot to do the push after committing test code.  Tests are there now, but can probably use some tweaking.  They concentrate on verifying minimal functionality instead of complete coverage.

m4olivei commented:
Addressed all review comments. Added some extra assertions, rebased against 8.x-2.x, and cleaned up some code style.

@deviantintegral could you give this another look?

deviantintegral commented:
The backslash thing is no big deal, whatever you decide this is good to merge. Thanks!

m4olivei commented:
Thanks!
